src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,619 @@
+package org.hydev.mcpm.client.database;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+import com.opencsv.exceptions.CsvException;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.hydev.mcpm.client.database.boundary.SearchPackagesBoundary;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesInput;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesType;
+import org.hydev.mcpm.client.database.results.SearchPackagesResult;
+import org.hydev.mcpm.client.models.PluginModel;
+import org.hydev.mcpm.client.models.PluginVersion;
+import org.hydev.mcpm.client.models.PluginYml;
+import org.hydev.mcpm.utils.PluginJarFile;
+
+import java.io.*;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * 
+ * Representation of a plugin for use in the local plugin tracker.
+ */
+
+class PluginTrackerModel {
+    String name;
+    Boolean isManual;
+    String versionId;
+    String pluginId;
+
+    public PluginTrackerModel(String name, Boolean isManual, String versionId, String pluginId) {
+        this.name = name;
+        this.isManual = isManual;
+        this.versionId = versionId;
+        this.pluginId = pluginId;
+    }
+
+    public PluginTrackerModel(String stringRepresentation) {
+        String[] split = stringRepresentation.split(",");
+        this.name = split[0];
+        this.isManual = Boolean.parseBoolean(split[1]);
+        this.versionId = split[2];
+        this.pluginId = split[3];
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Boolean getManual() {
+        return isManual;
+    }
+
+    public String getVersionId() {
+        return versionId;
+    }
+
+    public String getPluginId() {
+        return pluginId;
+    }
+}
+
+/**
+ * This class keeps track of locally installed packages
+ *
+ * @author Kevin (https://github.com/kchprog)
+ * @since 2022-09-27
+ */
+public class SuperLocalPluginTracker implements SuperPluginTracker {
+    // CSV file storing the list of manually installed plugins.
+    // Now, each row in the csv file represents something as follows:
+    // ""name", "boolean", "versionId", "pluginId"
+    private String mainLockFile = "plugins/mcpm.lock.csv";
+
+    // Directory containing the plugins
+    private String pluginDirectory = "plugins";
+
+    final ObjectMapper mapper = new ObjectMapper();
+
+    /*
```

Comment:
```suggestion
    /**
```
This should be javadoc lmao